adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2016 Red Hat
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Red Hat - initial creation
+ */
+
+package org.eclipse.hono.adapter.mqtt;
+
+import io.netty.handler.codec.mqtt.MqttConnectReturnCode;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.mqtt.MqttEndpoint;
+import io.vertx.mqtt.MqttServer;
+import io.vertx.mqtt.MqttServerOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * A Vert.x based Hono protocol adapter for accessing Hono's Telemetry &amp; Registration API using MQTT.
```

Comment:
I do not think that we should expose the Registration API via MQTT. I had only done that in the HTTP adapter because it is so convenient to be able to check registration data from the command line using curl or HTTPie ...
WDYT?